mvg/analysis_classes.py

```diff
@@ -48,13 +48,46 @@ def __init__(self, results, t_zone="Europe/Stockholm", t_unit="ms"):
             time unit for conversion from epoch time [ms].
         """
 
-        self.raw_results = results
-        self.t_zone = t_zone
-        self.t_unit = t_unit
-        self.dframe = None
-        self.inputs = results.get("inputs", "Inputs not available")
+        # raw_results as returned from server
+        self._raw_results = results
+
+        # timezone and unit set in constructor
+        self._t_zone = t_zone
+        self._t_unit = t_unit
+
+        # Dataframe representation
+        self._dframe = None
+
+        self._inputs = results.get("inputs", "Inputs not available")
         self.time_column = None
 
+    def _render_plot(self, interactive):
+        """Render plot to screen (interactive) or file.
+
+        Parameters
+        ---------
+        interactive: bool
+            Wheter to display plot on screen (True) or to store to file (False).
+
+        Returns
+        -------
+        plot file name: str
+          name of plot file (or emtpy string in case of interactive plot)
+        """
+        if interactive:
+            plot_file = ""
+            plt.show()
+        else:
+            if len(self.sources()) > 1:
+                srcstr = self.sources()[0] + "_to_" + self.sources()[-1] + "_"
+            else:
+                srcstr = self.sources()[0] + "_"
+            plot_file = srcstr + self.request_id() + ".png"
+            plt.savefig(plot_file, dpi=600, bbox_inches="tight")
+            print(f"saved plot to {plot_file}")
+
+        return plot_file
+
     def _add_datetime(self):
```

Comment:
This function should use `_add_datetime_df`, because now they contain nearly identical code

Reply:
Sure!